contracts/multisig-prover/src/payload.rs

```diff
@@ -19,16 +19,30 @@ pub enum Payload {
 
 impl Payload {
     /// id returns the unique identifier for the payload, which can be either
-    /// - the hash of comma separated sorted message ids
-    /// - the hash of the verifier set
+    /// - 0 followed by the hash of '_' separated message ids
+    /// - 1 followed by the hash of the verifier set
     pub fn id(&self) -> PayloadId {
-        match self {
+        let hash = match self {
             Payload::Messages(msgs) => {
-                let message_ids: Vec<_> = msgs.iter().map(|msg| msg.cc_id.clone()).collect();
+                let message_ids: Vec<String> =
+                    msgs.iter().map(|msg| msg.cc_id.to_string()).collect();
 
-                message_ids.as_slice().into()
+                Keccak256::digest(message_ids.join(&FIELD_DELIMITER.to_string()).as_bytes())
+                    .to_vec()
             }
-            Payload::VerifierSet(verifier_set) => verifier_set.hash().as_slice().into(),
+            Payload::VerifierSet(verifier_set) => verifier_set.hash().to_vec(),
+        };
+
+        let mut id = vec![self.variant_to_u8()];
+        id.extend(hash);
```

Comment:
shouldn't this be added to the value before it's hashed? Now the PayloadId is 33 bytes right?

Reply:
yeah, switched to double hashing instead since we decided to preserve 32 byte length